src/engine/QueryExecutionTree.cpp

```diff
@@ -480,50 +498,70 @@ QueryExecutionTree::generateRdfGraph(
 // _____________________________________________________________________________
 ad_utility::stream_generator::stream_generator
 QueryExecutionTree::writeRdfGraphTurtle(
-    const std::vector<std::array<VarOrTerm, 3>>& constructTriples, size_t limit,
-    size_t offset, std::shared_ptr<const ResultTable> res) const {
+    const Types::Triples& constructTriples, size_t limit, size_t offset,
+    std::shared_ptr<const ResultTable> res) const {
   auto generator = generateRdfGraph(constructTriples, limit, offset, res);
   for (const auto& triple : generator) {
     co_yield triple._subject;
     co_yield ' ';
-    co_yield triple._verb;
+    co_yield triple._predicate;
     co_yield ' ';
     co_yield triple._object;
     co_yield " .\n";
   }
 }
 
 // _____________________________________________________________________________
+template <QueryExecutionTree::ExportSubFormat format>
 ad_utility::stream_generator::stream_generator
 QueryExecutionTree::writeRdfGraphSeparatedValues(
-    const std::vector<std::array<VarOrTerm, 3>>& constructTriples, size_t limit,
-    size_t offset, std::shared_ptr<const ResultTable> res, char sep) const {
-  if (sep == 'b') {
+    const Types::Triples& constructTriples, size_t limit, size_t offset,
+    std::shared_ptr<const ResultTable> res) const {
+  if constexpr (format == BINARY) {
     throw std::runtime_error{
         "Binary export is not supported for CONSTRUCT queries"};
   }
-  const auto& escapeFunction =
-      sep == '\t' ? RdfEscaping::escapeForTsv : RdfEscaping::escapeForCsv;
+  constexpr auto& escapeFunction =
+      format == TSV ? RdfEscaping::escapeForTsv : RdfEscaping::escapeForCsv;
+  constexpr char sep = format == TSV ? '\t' : ',';
```

Comment:
you can even `static_assert` that the Enum is in Binary/Tsv/Csv, then we never forget something,
even if we add another format in this logic.